### PR TITLE
Solved Issue 1

### LINKED ISSUE #1: Train & Save Clean CIFAR-10 ResNet-18 test
## Summary
Train a clean ResNet-18 on CIFAR-10 and save the best checkpoint to models/resnet18_clean.pth. This model will be used to extract embeddings and validate the Spectral Signatures defense which we will implement in Sprint 2

**Acceptance Criteria**

-  [ ] models/resnet18_clean.pth saved.

-  [ ] Test accuracy ≥70% (or documented if lower due to timebox).

-  [ ] Log file contains: seed, hyperparams, train/val metrics, final test acc.

-  [ ] Loader test command runs successfully.

## Tasks that need to be completed for this feature
 - Add training script to: scripts/train_resnet18.py.

 - Script arguments(for reproducability): --epochs, --batch-size, --lr, --seed, --out models/resnet18_clean.pth.

For example: A script could be:
python3 scripts/train_resnet18.py --epochs 30 --batch-size 128 --lr 0.1 --seed 42 --out models/resnet18_clean.pth

 - Implement train/val split (e.g., 45k/5k).Save best model by val acc.

 - Log metrics to stdout and save final test accuracy to logs/train_clean.txt.

 - Save best checkpoint (by val acc) to models/resnet18_clean.pth.

 - Verify loader compatibility:
    python -c "from mithridatium.loader import load_resnet18; \
           load_resnet18('models/resnet18_clean.pth'); print('ok')"

## Additional Context

- Target accuracy: ≥70% on CIFAR-10 test is acceptable for baseline (timebox epochs; note the result).
- Use standard augmentation (random crop/flip) and ImageNet normalization (to match our evaluator transforms).
Example of Random crop:
import torchvision.transforms as transforms

transform_train = transforms.Compose([
    transforms.RandomCrop(32, padding=4),   # crop randomly within 4 pixels
    transforms.RandomHorizontalFlip(),      # flip left/right
    transforms.ToTensor(),
    transforms.Normalize((0.485, 0.456, 0.406),
                         (0.229, 0.224, 0.225))  # ImageNet stats
])

- Set a fixed seed for reproducibility and print it in logs.

### PR DESCRIPTION
**Pull request by williamphoenix**
## Description

Fixes #1

Wrote a script scripts/train_resnet18.py which takes training arguments to produce a resnet18 model trained on CIFAR10.

# 1. What was Changed

Running the training script allows the user to train a model and output a checkpoint at a desired or default location. Seed arguments are provided for reproducibility. Additions were made to the gitignore to avoid venvs and checkpoint files being pushed.

# 2. Why it was Changed

These changes allow clean models to be trained for testing.

# 3. How it was Changed

General training functions used via PyTorch. Torchvision's resnet18 used for general model architecture. Goal was to get model test accuracy to > 70% using a basic, easy to understand training script. Additional optimizations can be made if model accuracy is too low, but my testing could typically get the model to converge at ~75% accuracy within ~60 epochs.

### CODE REVIEW

File: scripts/train_resnet18.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        for x, y in train_loader:
+            x, y = x.to(device), y.to(device)
+            optimizer.zero_grad(set_to_none=True)
+            loss = criterion(model(x), y)
+            loss.backward()
+            optimizer.step()
+        val_loss, val_acc = evaluate(model, test_loader, device, criterion)
+        print(f"Epoch {epoch+1}/{epochs} - val_loss: {val_loss:.4f}  val_acc: {val_acc:.3f}")
+
+    torch.save(model.state_dict(), args.output_path)
```
Comment (pelumitegbe) [inline scripts/train_resnet18.py:77]: Right now the script only saves the raw state_dict once at the end. It might be useful to also save the best checkpoint during training (based on validation accuracy). This helps avoid cases where later epochs overfit or diverge. A validation split (e.g. 45k train / 5k val) plus “save best” logic would make the training script more robust.


best_acc = 0.0
for epoch in range(epochs):
    # training loop...
    val_loss, val_acc = evaluate(model, val_loader, device, criterion)
    if val_acc > best_acc:
        best_acc = val_acc
        torch.save(model.state_dict(), args.output_path)
        print(f"Saved best model with acc={best_acc:.2f}")
